### ISSUE TITLE
[Errno 2] could not open alignment file `reference_all_to_target_all.sam`: No such file or directory

### ISSUE THREAD

**Issue by charlesfeigin** [agshumate/Liftoff#12]
Hi, I just tried using genelift to lift over annotations between assemblies for two closely related species. However, the pipeline crashes at a fairly early state with the following error:

2020-06-27 13:41:27,516 - INFO - Populating features
2020-06-27 13:45:44,661 - INFO - Populating features table and first-order relations: 1327486 features
2020-06-27 13:45:44,661 - INFO - Updating relations
2020-06-27 13:46:04,737 - INFO - Creating relations(parent) index
2020-06-27 13:46:06,371 - INFO - Creating relations(child) index
2020-06-27 13:46:08,453 - INFO - Creating features(featuretype) index
2020-06-27 13:46:09,563 - INFO - Creating features (seqid, start, end) index
2020-06-27 13:46:11,139 - INFO - Creating features (seqid, start, end, strand) index
2020-06-27 13:46:12,867 - INFO - Running ANALYZE features
[E::hts_open_format] Failed to open file "genelift_test1_intermediates/reference_all_to_target_all.sam" : No such file or directory
Traceback (most recent call last):
  File "liftoff/liftoff.py", line 121, in <module>
    main()
  File "liftoff/liftoff.py", line 74, in main
    unmapped_features, infer_transcripts, infer_genes, cov_threshold, seq_threshold, minimap2_path, inter_files)
  File "/path/to/working/directory/liftoff/liftoff/liftover_types.py", line 17, in lift_original_annotation
    unmapped_features, reference_fasta, minimap2_path, inter_files, True)
  File "/path/to/working/directory/liftoff/liftoff/align_features.py", line 29, in align_features_to_target
    aligned_segments = parse_alignment(file, parent_dict, children_dict, unmapped_features, search_type)
  File "/path/to/working/directory/liftoff/liftoff/align_features.py", line 91, in parse_alignment
    sam_file = pysam.AlignmentFile(file,'r',check_sq=False, check_header=False)
  File "pysam/libcalignmentfile.pyx", line 742, in pysam.libcalignmentfile.AlignmentFile.__cinit__
  File "pysam/libcalignmentfile.pyx", line 941, in pysam.libcalignmentfile.AlignmentFile._open
FileNotFoundError: [Errno 2] could not open alignment file `genelift_test1_intermediates/reference_all_to_target_all.sam`: No such file or directory

minimap2 and pysam are both installed and in my path in the conda environment I created. reference_all_genes.fa was produced and is a correctly-formatted multifasta of genes, however It appears that the sam file was never created. Do you have any idea what may be wrong here? Thanks!

**Comment by agshumate** [comment]
hi can you try re-installing the current version of Liftoff and let me know if the problem persists?

**Comment by charlesfeigin** [comment]
Hi, thanks so much for your fast reply. I actually tried this between posting the original issue and your reply with the same outcome :(

**Comment by agshumate** [comment]
what version of Minimap2 are you using? Is it possible you could provide your data for me to test?

**Comment by charlesfeigin** [comment]
Minimap2 version: 2.17-r941

Unfortunately this is a collaborator's data that I can't share. However, this lead me to download a very similar dataset to try and replicate the error. Oddly, this didn't fail and produced a .sam output, suggesting something was up with my inputs. I compared my files vs the test dataset and the only significant difference I could find was that my target genome fasta had 60 line characters, whereas all of the others had 80. I changed the line width to 80 and now it produced a .sam output too. Unexpected fix, but thank so much for your response!
